fairtools/MCConfigurator/FairYamlVMCConfig.cxx

```diff
@@ -183,22 +190,25 @@ void FairYamlVMCConfig::SetupGeant4()
         //        LOG(info) << "Setting Geant4 multithreaded to " << (mtMode?"true":"false");
     }
 
-    FairFastSimRunConfiguration* runConfiguration =
-        new FairFastSimRunConfiguration(fYamlConfig["Geant4_UserGeometry"].as<std::string>(),
-                                        fYamlConfig["Geant4_PhysicsList"].as<std::string>(),
-                                        fYamlConfig["Geant4_SpecialProcess"].as<std::string>(),
-                                        specialStacking,
-                                        mtMode);
+    auto const useFastSim(fYamlConfig["UseFastSim"] ? fYamlConfig["UseFastSim"].as<bool>() : fUseFastSimDefault);
+    std::unique_ptr<TG4RunConfiguration> runConfiguration;
+    if (useFastSim) {
+        runConfiguration = std::make_unique<FairFastSimRunConfiguration>(
+            g4UserGeometry, g4PhysicsList, g4SpecialProcess, specialStacking, mtMode);
+    } else {
+        runConfiguration = std::make_unique<TG4RunConfiguration>(
+            g4UserGeometry, g4PhysicsList, g4SpecialProcess, specialStacking, mtMode);
+    }
 
-    TGeant4* geant4 = new TGeant4("TGeant4", "The Geant4 Monte Carlo", runConfiguration);
+    TGeant4* geant4 = new TGeant4("TGeant4", "The Geant4 Monte Carlo", runConfiguration.release());
```

Comment:
Just wondering: Where is this pointer (ownership) going after the method ends?

(Not the target of this PR, I know!)

Reply:
https://github.com/vmc-project/geant4_vmc/blob/f6ced3fa4a5253848186c466b4865f37e75d223e/source/run/src/TGeant4.cxx#L165-L169

I guess, one would call this a "conditional singleton"? TBH, since a few years, when I got aware of this *pattern*, which is btw also used throughout the FairRoot steering classes, I could not find a good idea yet how to fix this situation.

Reply:
Thanks for the explanation!

Fixing this without a breaking change is really, really non-trivial. I agree.

Reply:
The user interface of this singleton implementation is also always confusing me. Because it sort of allows you to construct the object, but implicitly requires you to instantiate as a heap object. And it also requires you to first call a constructor before you get what you want from calling [`TGeant4* TGeant4::MasterInstance()`](https://github.com/vmc-project/geant4_vmc/blob/f6ced3fa4a5253848186c466b4865f37e75d223e/source/run/include/TGeant4.icc#L26). I have also seen macros with just a single line `new FairRootManager;` which completely blew my mind.

Reply:
I know, this is not the target of this PR!

Could you consider adding a comment above this line? Something like
```c++
// Instantiate a singleton like object, "leaking" it is the current API
```
Rephrase / Modify to your liking!

Just something, so that people like me now that this is not a leak.

P.S.: I am not going to ask, whether FairRoot should call `TGeant4::MasterInstance()` to make sure that the singleton like thing isn't already initialized.

Reply:
done